### ISSUE TITLE
Why doesn't "variables" return new variables after refetch?

### ISSUE THREAD

**Issue by cksal0805** [apollographql/apollo-client#9236]
Why doesn't "variables" return new variables after refetch?

This is the code I wrote.
```js
const { data, loading, variables, refetch } = useQuery(SAMPLE_QUERY, {
    fetchPolicy: 'cache-and-network',
    variables: {
      where: {
        isUsed: true,
      },
    },
});
```

When I click the button, I run it with new variables in the refetch.
```jsx
<button onClick={() => refetch({ where: {isUsed: false } })} />
```

```js
console.log(variables) // where : { isUsed: true } 
```
Refetched, but variables returned from useQuery retain the initial value.
In other words, it will not be updated.

Why doesn't "variables" return new variables after refetch?
I wonder if this is a bug or an intention.

If it is the intended logic, what is the role of variables returning from useQuery?
```js
const {  variables } = useQuery(SAMPLE_QUERY) 
```

**Comment by hwillson** [comment]
Interesting - this doesn't appear to be intentional or a bug; it looks more like a use case that just isn't covered. We'll investigate further, thanks!

**Comment by cksal0805** [comment]
Thank you!
I want to control the management of variables with variables returned from useQuery instead of state and setState.

**Comment by zmen** [comment]
> Interesting - this doesn't appear to be intentional or a bug; it looks more like a use case that just isn't covered. We'll investigate further, thanks!

Seems this behavior is brought by commit `b8ec4660fde2cb9bb5046a43c15d20eaec3e3b2b` which is related with #9135. `variables` can be updated correctly before that commit.

**Comment by cksal0805** [comment]
@zmen Yes, the issue is not reproduced in version apollo-client 3.4.17 + graphql 15.

**Comment by ludusrusso** [comment]
any news about this?

**Comment by cksal0805** [comment]
@ludusrusso Now, The issue is not reproduced in version apollo-client 3.4.17 + graphql 15.
I don't know if the new version has solved it.

**Comment by ab-pm** [comment]
`fetchMore` has the same problem. This is very annoying when doing pagination, but not knowing how many elements have been requested.
